File: LeetCode-Solved-Problems/0067-add-binary/0067-add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int c=0;
        int la=a.length()-1,lb=b.length()-1;
        string res;
        while(la>=0 || lb>=0){
            int sum=c;
            if(la>=0)sum+=a[la--]-'0';
            if(lb>=0)sum+=b[lb--]-'0';
            
            c = sum>1?1:0;
            res+=to_string(sum%2);
        }
        
        if(c)res+=to_string(c);
        reverse(res.begin(),res.end());
        
        return res;
        
    }
};

File: LeetCode-Solved-Problems/0067-add-binary/0067-add-binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        res=""
        la,lb,c = len(a)-1 , len(b)-1 ,0
        while la>=0 or lb>=0:
            s = c
            if la>=0:s+=ord(a[la])-ord('0') # ord is use to get value of ASCII character
            if lb>=0:s+=ord(b[lb])-ord('0')
            la,lb=la-1,lb-1
            c=1 if s>1 else 0
            res+= str(s%2)
            
        if c != 0:res+=str(c)
        return res[::-1]